doc/pages/Manual.md

```diff
@@ -698,3 +866,248 @@ recommended to check the validity and quality of your `opam` files.
+  [`pre-install-commands:`](#configfield-pre-install-commands),
+  [`pre-remove-commands:`](#configfield-pre-remove-commands):
+  as the corresponding [global config](#config) fields.
+- [`wrap-build-commands:`](#configfield-pre-build-commands),
+  [`wrap-install-commands:`](#configfield-pre-install-commands),
+  [`wrap-remove-commands:`](#configfield-pre-remove-commands):
+  as the corresponding [global config](#config) fields.
+- [`post-build-commands:`](#configfield-pre-build-commands),
+  [`post-install-commands:`](#configfield-pre-install-commands),
+  [`post-remove-commands:`](#configfield-pre-remove-commands):
+  as the corresponding [global config](#config) fields.
+- <a id="switchconfigsection-paths">`paths: "{" { <ident>: <string> ... } "}"`</a>:
+  defines the standard paths within the switch: recognised fields include
+  `prefix:`, `bin:`, `sbin:`, `lib:`, `share:`, `etc:`, `doc:`, `man:`,
+  `stublibs:`, `toplevel:`.
+- <a id="switchconfigsection-variables">`variables: "{" { <ident>: <string> ... } "}"`</a>:
+  allows the definition of variables local to the switch.
+
+#### switch-state
+
+This file, located at `<switch-prefix>/.opam-switch/switch-state`, is used by
+opam to store the current state of a switch. All fields are lists of packages
+specified as strings in the format `<package>.<version>`.
+
+- <a id="switchstatefield-opam-version">`opam-version: <string>`</a>:
+  the file format version.
+- <a id="switchstatefield-compiler">`compiler: [ <string> ... ]`</a>:
+  the package that form the base of the current switch. They won't be affected
+  by `opam upgrade`, and are protected against deletion.
+- <a id="switchstatefield-roots">`roots: [ <string> ... ]`</a>:
+  lists the package "roots", _i.e._ packages that have been installed manually
+  by the user (as opposed to the ones which got installed to fulfil
+  dependencies). Note that packages in this list may not be installed, if they
+  were removed due to build failures.
+- <a id="switchstatefield-installed">`installed: [ <string> ... ]`</a>:
+  lists all currently installed packages.
+- <a id="switchstatefield-pinned">`pinned: [ <string> ... ]`</a>:
+  lists all packages which have been pinned, and to what version. If an overlay
+  package definition is present in
+  `<switch-prefix>/.opam-switch/overlay/<pkgname>/`, that can be used to alter
+  the package definition and modify its source URL. Otherwise, the package is
+  simply pinned to the given version.
+
+#### repos-config
+
+This file is at `~/.opam/repo/repos-config` and lists all package repositories
+configured on the system, and their source URLs.
+
+- <a id="reposconfigfield-repositories">`repositories: [ <string> { <URL> } ... ]`</a>:
+  lists the configured repository idents and their URLs.
+
+Note that repositories without URLs can be used. In that case, the data in the
+corresponding directory will be used but never updated by opam ; also, caching
```

Comment:
It can be used for directly testing and hot-patching local repositories, or have package overlays specific to your opam root. Yes, this is kind of tautological in this case, opam doesn't update since it doesn't have a source to update from, but the fact that the internal directory is only used read-only by opam is worth mentioning.
This is mainly directed towards experiments, at the moment, there is no UI to handle that.